cirq/experiments/fidelity_estimation.py

```diff
@@ -644,10 +773,29 @@ def _summary_stats(row):
     df = df.apply(_summary_stats, axis=1)
 
     def per_cycle_depth(df):
+        """This function is applied per cycle_depth in the following groupby aggregation."""
         fid_lsq = df['numerator'].sum() / df['denominator'].sum()
-        return pd.Series({'fidelity': fid_lsq})
+        ret = {'fidelity': fid_lsq}
+
+        def _try_keep(k):
+            """If all the values for a key `k` are the same in this group, we can keep it."""
+            if k not in df.columns:
+                return  # coverage: ignore
+            vals = df[k].unique()
+            if len(vals) == 1:
+                ret[k] = vals[0]
+
+        _try_keep('q0')
+        _try_keep('q1')
+        _try_keep('pair_name')
```

Comment:
Shouldn't that be rather a failure? (When they're not the same)

Reply:
yes, I think you're right. Maybe this leniency was a holdover from previous design? changing to error. edit: ah yes, I was going to just run it on every column instead of hand-writing the ones to keep. But changed my mind

Reply:
Hmm, so what's the final outcome on this?

Reply:
it will preserve the keys listed, but a) doesn't care if they're not present as they are metadata b) makes sure the grouped values in a given column are all compatible.

I think it's a good idea to keep this around if and when we add additional metadata